.github/workflows/build2.yml

```diff
@@ -0,0 +1,85 @@
+name: build2
+
+on: [push, pull_request]
+
+jobs:
+
+  # Job for building on Ubuntu
+  Ubuntu:
+    name: ${{matrix.config.name}}
+    runs-on: ${{matrix.config.os}}
+    strategy:
+      matrix:
+        config:
+          - name: Ubuntu gcc
+            os: ubuntu-latest
+            compiler_cc: gcc
+            compiler_cxx: g++
+          - name: Ubuntu clang
+            os: ubuntu-latest
+            compiler_cc: clang
+            compiler_cxx: clang++
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - uses: build2/setup-build2-github-action@v2
+        with:
+          build2-version: staged
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get install -y libx11-dev libxrandr-dev libudev-dev libglu1-mesa-dev freeglut3-dev mesa-common-dev xorg-dev
+      - name: Configure
+        run: bdep init -C @${{matrix.config.compiler_cc}} cc config.cxx=${{matrix.config.compiler_cxx}}
```

Comment:
Good point, I was thinking about Debug vs. Release mode, too. Now, the question is what would you consider a Release vs. Debug build? I mean build2 does not (yet) provide predefined build modes, afaik. Would you just compile with `-O3 -DNDEBUG` for Release? What would the equivalent on MSVC be? Is there anything I should consider for MacOS?

Reply:
Here is the `build2` ticket about potential pre-defined configurations (that will not come soon, to be clear), I think it's a good basis: https://github.com/build2/build2/issues/11 Although it doesnt talk about MSVC.

> Would you just compile with -O3 -DNDEBUG for Release?

That's more or less the idea, (although I would use -O2 by default personally, but watever XD). Just "optimize please" flags for Release, and "enable debugging data so that I can debug that" for Debug.
No need to add more test configurations. 

I usually go with:
- msvc-release: 
```
config.cc.coptions += /O2
```
- msvc-debug (notice that `/MDd` is in the compiler name): 
```
config.cxx=cl /MDd
config.cc.coptions += /Od /Zi
config.cc.loptions += /DEBUG
```
- gcc/clang/emscripten-release:
```
config.cc.coptions += -O2
```
- gcc/clang/emscripten-debug (`O0` could be `Og`, but I don't know the differences):
```
config.cc.coptions += -g -O0
config.cc.loptions += -DEBUG
```
(note: emscripten could have an additional flag, I'll check when we work on it)